pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1265,26 +1281,53 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 
 	volumePlugin, err := og.volumePluginMgr.FindExpandablePluginBySpec(volumeSpec)
 
-	if err != nil {
+	if err != nil && volumeSpec.IsControllerExpandable() {
 		return volumetypes.GeneratedOperations{}, fmt.Errorf("Error finding plugin for expanding volume: %q with error %v", pvcWithResizeRequest.QualifiedName(), err)
+	} else if !volumeSpec.IsControllerExpandable() {
+		// A flex volume is expandable but since it resides on the kubelet not the controller
+		// and is dynamically probed plugin, FindExpandablePluginBySpec will never find it
+		// Therefore fake the volume expansion on the controller.
+		glog.V(4).Infof("Faking expansion for flex volume %s", pvcWithResizeRequest.QualifiedName())
 	}
-	if volumePlugin == nil {
+
+	if volumePlugin == nil && !volumeSpec.IsControllerExpandable() {
 		return volumetypes.GeneratedOperations{}, fmt.Errorf("Can not find plugin for expanding volume: %q", pvcWithResizeRequest.QualifiedName())
 	}
```

Comment:
Can this entire if-else be solved by a instance of flexvolume plugin that is returned via `FindExapndablePlugin` call and performs a no-op resize when called and returns with success response?

Reply:
@gnufied to install a fake/no-op flexvolume plugin on the controller that will be found via `FindExpandablePlugin` for any driver name in the Spec? If we were to do that we'd have to do something like `if Spec.IsKubeletExpandable() { return noopExpandableInstance{} }` 

If that's fine I will make that change.